elasticsearch/src/test/java/org/apache/calcite/adapter/elasticsearch/EmbeddedElasticsearchPolicy.java

```diff
@@ -216,4 +209,11 @@ private TransportAddress httpAddress() {
     return node.httpAddress();
   }
 
+  @Override public void afterAll(ExtensionContext context) throws Exception {
```

Comment:
Just a small note: could you please move the methods to the location of previous `before` and `after`?

Reply:
Well , Ok Let me change it.  In addition, can we add style to the calite project to define the code format in calite?

Reply:
What do you mean by "code format"?

I'm inclined that we should prefer https://github.com/apache/calcite/blob/master/.editorconfig when possible.

Reply:
PS. please move `afterAll` as well

Reply:
hi @vlsi  Is your original idea to implement declarative registration extensions and kotlin?

Reply:
No, I just had no idea how to approach the test.

The approach you use seems to be nice. I have not checked TestContainer best practices yet. They have a case for "keeping test container up for multiple test classes", so there might be something to learn from.

Reply:
> No, I just had no idea how to approach the test.
> 
> The approach you use seems to be nice. I have not checked TestContainer best practices yet. They have a case for "keeping test container up for multiple test classes", so there might be something to learn from.

I am also interested in learning about "keeping test container up for multiple test classes".

Reply:
It looks like JUnit provides no solution yet, so TestContainers use their own: https://github.com/testcontainers/testcontainers-java/pull/887/files#diff-0dcf4d2cadeb3e526020477bc12e1420R73

Reply:
> It looks like JUnit provides no solution yet, so TestContainers use their own: https://github.com/testcontainers/testcontainers-java/pull/887/files#diff-0dcf4d2cadeb3e526020477bc12e1420R73

Well, indeed, TestContainers also looks good. Thank you very much for letting me learn something new.